pkg/apis/resource/types.go

```diff
@@ -129,106 +644,142 @@ type ResourceClaimStatus struct {
+	// devices in the result because it is up to the drivers which configuration
+	// parameters they support. They can silently ignore unknown configuration
+	// parameters.
+	//
+	// +optional
+	// +listType=atomic
+	Config []DeviceAllocationConfiguration
 }
 
-// ResourceHandleDataMaxSize represents the maximum size of resourceHandle.data.
-const ResourceHandleDataMaxSize = 16 * 1024
+// AllocationResultsMaxSize represents the maximum number of
+// entries in allocation.devices.results.
+const AllocationResultsMaxSize = 32
 
-// StructuredResourceHandle is the in-tree representation of the allocation result.
-type StructuredResourceHandle struct {
-	// VendorClassParameters are the per-claim configuration parameters
-	// from the resource class at the time that the claim was allocated.
-	VendorClassParameters runtime.Object
+// DeviceRequestAllocationResult contains the allocation result for one request.
+type DeviceRequestAllocationResult struct {
+	// Request is the name of the request in the claim which caused this
+	// device to be allocated. Multiple devices may have been allocated
+	// per request.
+	//
+	// +required
+	Request string
 
-	// VendorClaimParameters are the per-claim configuration parameters
-	// from the resource claim parameters at the time that the claim was
-	// allocated.
-	VendorClaimParameters runtime.Object
+	// Driver specifies the name of the DRA driver whose kubelet
+	// plugin should be invoked to process the allocation once the claim is
+	// needed on a node.
+	//
+	// Must be a DNS subdomain and should end with a DNS domain owned by the
+	// vendor of the driver.
+	//
+	// +required
+	Driver string
 
-	// NodeName is the name of the node providing the necessary resources
-	// if the resources are local to a node.
-	NodeName string
+	// This name together with the driver name and the device name field
+	// identify which device was allocated (`<driver name>/<pool name>/<device name>`).
+	//
+	// Must not be longer than 253 characters and may contain one or more
+	// DNS sub-domains separated by slashes.
+	//
+	// +required
+	Pool string
 
-	// Results lists all allocated driver resources.
-	Results []DriverAllocationResult
+	// Device references one device instance via its name in the driver's
+	// resource pool. It must be a DNS label.
+	//
+	// +required
+	Device string
 }
 
-// DriverAllocationResult contains vendor parameters and the allocation result for
-// one request.
-type DriverAllocationResult struct {
-	// VendorRequestParameters are the per-request configuration parameters
-	// from the time that the claim was allocated.
-	VendorRequestParameters runtime.Object
+// DeviceAllocationConfiguration gets embedded in an AllocationResult.
+type DeviceAllocationConfiguration struct {
+	// Source records whether the configuration comes from a class and thus
+	// is not something that a normal user would have been able to set
+	// or from a claim.
+	//
+	// +required
+	Source AllocationConfigSource
```

Comment:
There was some discussion on removing config from a class, but I don't remember the conclusion there. If the conclusion was to remove it -- is this then here to allow us to reintroduce it in the future?

Reply:
`DeviceClass` has config, and this is how the driver can tell where a configuration came from.